Assets/Scripts/Visuals/ContextMenuVisuals.cs

```diff
@@ -143,28 +148,48 @@ void CreateSingleMenu(ContextMenuState.Data menuData)
                             var overshoot = bottomRightCorner.x - Screen.width;
                             pos = new Vector3(pos.x - overshoot, pos.y, pos.z);
                         }
+
                         break;
                     case ContextMenuState.Placement.Direction.Left:
                         if (pos.x - width < 0)
                         {
                             continue;
                         }
+
                         pos = new Vector3(pos.x - width, pos.y, pos.z);
                         break;
                     case ContextMenuState.Placement.Direction.Right:
                         if (bottomRightCorner.x > Screen.width)
                         {
                             continue;
                         }
+
                         break;
                     default:
                         break;
                 }
+
                 menuRect.position = pos;
                 break;
             }
         }
 
+        private static bool CheckAllSubItemsDisabled(List<ContextMenuItem> contextMenuItems)
+        {
+            foreach (var item in contextMenuItems)
+            {
+                switch (item)
+                {
+                    case ContextMenuTextItem {isDisabled: false}:
+                        return false;
+                    case ContextSubmenuItem {isDisabled: false}: 
```

Comment:
`case ContextSubmenuItem _:` should also work. Because `isDisabled` is never set to true

Reply:
Honestly I never used this syntax but it just assumes the `item` is a `contextmenutextitem`/`contextsubmenuitem` and checks if `isDisabled` is `false `and if so returns `false`. So nothing gets `set` here (?). At the moment the only place where `isDisabled` gets `set` to `true` is in `UIHierarchyVisuals`. I don't really understand. Is your suggestion the default case?

Reply:
Should I do this and remove the `isDisabled` from the state (line 84) or should I instead keep it in the SubMenuItemState and assign it instead of passing it in line 108 of `ContextMenuVisuals` ?

Reply:
I think, removing the `isDisabled` from the state is cleaner, because it is not really a state. Rather, it is fully dependent on other states (the states of the child items)